src/Options/Generic.hs

```diff
@@ -929,3 +948,20 @@ unwrapRecordPure
     -- ^ Command-line arguments
     -> Maybe (f Unwrapped)
 unwrapRecordPure = fmap unwrap . getRecordPure
+
+showHelpText :: Options.ParserPrefs -> Options.ParserInfo a -> IO ()
+showHelpText pprefs pinfo =
+  Options.handleParseResult . Options.Failure $
+  Options.parserFailure pprefs pinfo Options.ShowHelpText mempty
+
+-- | Marshal any value that implements 'ParseRecord' from the command line
+-- and unwrap its fields alongside an io action to print the help message
+unwrapWithHelp
+    :: (Functor io, MonadIO io, ParseRecord (f Wrapped), Unwrappable f)
+    => Text
+    -- ^ Program description
+    -> io (f Unwrapped, io ())
+    -- ^ (options, io action to print help message)
+unwrapWithHelp desc = do
+  (opts, help) <- getWithHelp desc
+  pure (unwrap opts, help)
```

Comment:
Minor suggestion: If you use `return` here you can avoid the extra `Functor` constraint